index.js

```diff
@@ -104,14 +106,14 @@ SocksProxyAgent.prototype.callback = function connect(req, opts, fn) {
       opts.port = null;
       s = tls.connect(opts);
     }
-    socket.resume();
```

Comment:
Is this not necessary anymore?

Reply:
That's right, directly from the breaking changes/migration guide:

> Sockets are no longer paused after a SOCKS connection is made, so socket.resume() is no longer required. (Please be sure to attach data handlers immediately to the Socket to avoid losing data).

source: https://github.com/JoshGlazebrook/socks/blob/master/docs/migratingFromV1.md

Both your tests and the `tor-request` tests are passing with the new versions and these changes ( I made sure they weren't using old dependencies/code )

Node v4 and Node v5 will not work though since only Node v6.0+ is supported by `socks` v2.

Reply:
Yes v2 does not pause the socket before it is passed back. As long as you attach a .on('data') handler immediately no data is lost.